lib/mfx/api.py

```diff
@@ -0,0 +1,141 @@
+  level   = property(lambda s: s.ifprop("level", "CodecLevel={level}"))
+  gop     = property(lambda s: s.ifprop("gop", "GopPicSize={gop}"))
+  bframes = property(lambda s: s.ifprop("bframes", "GopRefDist={bframes}"))
+  slices  = property(lambda s: s.ifprop("slices", "NumSlice={slices}"))
+  minrate = property(lambda s: s.ifprop("minrate", "TargetKbps={minrate}"))
+  maxrate = property(lambda s: s.ifprop("maxrate", "MaxKbps={maxrate}"))
+  refs    = property(lambda s: s.ifprop("refs", "NumRefFrame={refs}"))
+  ladepth = property(lambda s: s.ifprop("ladepth", "LookAheadDepth={ladepth}"))
+  extbrc  = property(lambda s: s.ifprop("extbrc", "ExtBRC={extbrc}"))
+  ldb     = property(lambda s: s.ifprop("ldb", "mfxExtCodingOption3.LowDelayBRC={ldb}"))
+  strict  = property(lambda s: s.ifprop("strict", "GopOptFlag={strict}"))
+  pict    = property(lambda s: s.ifprop("vpict", "mfxExtCodingOption.PicTimingSEI=0"))
+  quality = property(lambda s: s.ifprop("quality", "TargetUsage={quality}"))
+  lowpower  = property(lambda s: s.ifprop("lowpower", "LowPower={lowpower}"))
+  tilecols  = property(lambda s: s.ifprop("tilecols", "mfxExtAV1TileParam.NumTileColumns={tilecols}"))
+  tilerows  = property(lambda s: s.ifprop("tilerows", "mfxExtAV1TileParam.NumTileRows={tilerows}"))
+  maxframesize  = property(lambda s: s.ifprop("maxframesize", "MaxSliceSize={maxframesize}"))
+
+  @property
+  def profile(self):
+    def inner(profile):
+      return f"CodecProfile={self.map_profile(self.codec, self.props['profile'])}"
+    return self.ifprop("profile", inner)
+
+  @property
+  def rcmode(self):
+    return f"RateControlMethod={self.map_rcmode(self.props['rcmode'])}"
+
+  @property
+  def level(self):
+    def inner(level):
+      # TODO: need to define map_level
+      return f"CodecLevel={self.map_level(self.props['level'])}"
+    return self.ifprop("level", inner)
+
+  @property
+  def qp(self):
+    def inner(qp):
+      rcmode = self.props["rcmode"].upper()
+      mqp = qp
+      if self.codec in [Codec.MPEG2]:
+        mqp = mapRangeInt(qp, [0, 100], [1, 51])
+      elif self.codec in [Codec.AV1] and "ICQ" == rcmode:
+        mqp = mapRangeInt(qp, [0, 255], [1, 51])
+      elif self.codec in [Codec.HEVC] and "QVBR" == rcmode:
+        mqp = mapRangeInt(qp, [0, 255], [1, 51])
+      return f"QPI={mqp}:QPP={mqp}:QPB={mqp}"
+    return self.ifprop("qp", inner)
+
+  @property
+  def encparams(self):
+    params = [
+      self.profile, self.rcmode, self.qp, self.level, self.gop, self.bframes,
+      self.slices, self.minrate, self.maxrate, self.refs, self.ladepth,
+      self.extbrc, self.ldb, self.strict, self.pict, self.quality, self.lowpower,
+      self.tilecols, self.tilerows, self.maxframesize,
+    ]
+    return ':'.join(v for v in params if len(v) != 0)
+
+  @classmethod
+  @memoize
+  def map_profile(cls, codec, profile):
+    # TODO: we could move this method to the enum class as "lookup"
+    # then call MfxCodecProfile.lookup(codec, profile)
```

Comment:
Please resolve this TODO.

Reply:
the next patch will move map_profile  to the enum class as lookup